ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -145,6 +148,26 @@ LocalStream getLocal() {
     public String toString() {
       return JavaUtils.getClassSimpleName(getClass()) + ":" + request;
     }
+
+    private Set<RaftPeer> getSuccessors(RaftPeerId peerId) throws IOException {
+      final RoutingTable routingTable = request.getRoutingTable();
+      final RaftGroupId groupId = request.getRaftGroupId();
+      final RaftConfiguration conf = server.getDivision(groupId).getRaftConf();
+      Set<RaftPeer> successors = routingTable.getSuccessors(peerId).stream().map(conf::getPeer)
+          .collect(Collectors.toSet());
+      if (successors.size() > 0) {
```

Comment:
If successors.size() == 0, either there is no routingTable, or there is no successors in the route.  We need to handle this two cases separately.

When there is no routing table, we should change request.getRoutingTable() to return null.